### ISSUE TITLE
Update with important fixes

### ISSUE THREAD

**Issue by MyRoGeertGit** [xlnt-community/xlnt#18]
Thank you all for this new effort.

In https://github.com/tfussell/xlnt/issues/748#issuecomment-2282840153 @doomlar mentioned 3 important issues. As far as I can see, they're not included in this community release. 
My attention has been drawn to it by issue https://github.com/tfussell/xlnt/pull/686/commits/1eabb380b8151f1bb8557532d16590413ef3f6e0.
I would appreciate them.

Geert.

**Comment by doomlaur** [comment]
I also discussed this in #9 with @m7913d. Personally, I had an issue with certain files that caused XLNT exceptions. So we would have 2 possibilities:
1. Fix the exception I had by fixing issue #685.
2. Add full support for defined names (like #686 does, but eventually with some further improvements).

Before we can decide what is the best way to go, I would have the following questions:
- What is your use case, exactly?
- What file(s) are you trying to load or create? 
- Would your issue be fixed by fixing certain exceptions (option 1), or do you need full support for defined names (option 2)?

**Comment by MyRoGeertGit** [comment]
Our goal is to read spreadsheets files from xlsx and to write simple spreadsheets.
The spreadsheets are produced by other external platforms and contain typically database data. Important is that definedNames which users may be added aren't aborting reading of the spreadsheet, which is the case, if a name is defined on the workbook level. Making localSheetId optional is already sufficient for the moment. Your implementation #686 goes already much further, by implementing other optional keywords. Fixing the certain exceptions is sufficient. The following fix is what I needed.

```
@@ -2083,7 +2083,10 @@ void xlsx_consumer::read_office_document(const std::string &content_type) // CT_

                defined_name name;
                name.name = parser().attribute("name");
                {+if (parser().attribute_present("localSheetId"))+}
{+                {+}
                    name.sheet_id = parser().attribute<std::size_t>("localSheetId");
                {+}+}
                name.hidden = false;
```

**Comment by doomlaur** [comment]
The fix you are proposing makes sense. I will open a Pull Request tomorrow containing the fix. While I think that full support for `definedNames` might be desirable, for now it's important that XLNT does not throw unnecessary exceptions, so implementing your proposed fix is a step in the right direction :wink:

**Comment by doomlaur** [comment]
@MyRoGeertGit Your issue has been fixed by Pull Request #20. Please let us know if you find any other issues! :slightly_smiling_face:
